### ISSUE TITLE
Imported type incorrectly allowed to be used as a value when object shorthand is used

### ISSUE THREAD

**Issue by samouri** [microsoft/TypeScript#44868]
# Bug Report

Normally, TS will figure out if an imported type is incorrectly being used as a value.
When shorthand object creation is used, TS will miss the error.

### ⏯ Playground Link

I couldn't figure out how to use the playground since it depends on using multiple files (import/export).
I've created a minimal repro in this repo [samouri/typescript-type-import-bug-repro](https://github.com/samouri/typescript-type-import-bug-repro), it is only ~10 LOC.

### 💻 Code

```ts
import type { A } from './a';

// This correctly throws
const object1 = {A: A}

// This should throw the error: "'A' cannot be used as a value because it was imported using 'import type'"
const object2 = { A };
```

### 🙁 Actual behavior

No error is thrown

### 🙂 Expected behavior

An error should be thrown. Types should not be used where values are expected.

**Comment by orta** [comment]
Strange, this seems to raise `2693` correctly in the TS bug workbench

```ts repro
// @errors: 2693
// @filename: other.ts
export type A = {
    thing: string
}
// @filename: index.ts
import type { A } from './other';

// This should throw the error: "'A' cannot be used as a value because it was imported using 'import type'"
const object = { A }; 
```

[Workbench Repro](https://www.typescriptlang.org/dev/bug-workbench/?#code/PTAEAEFMCdoe2gZwFygEwDYCcBmAsAFAgQBmAlgDaQB2AhgLaSpwAuAFjAHQuKGQAeABwQtQLAJ6DIoAIKgAvKADehUGrFsy1AOapELaFu2EAvoWLhyVOo1RaAJgO68CZesOiiJU5bNAnQEnh6UAByTmBWDmhQgG5CczAAFU1EUEQ2OABXCnsNeAB3DWkYeGhUACJQmVDQAGNaampWUAAjaSzESDzaNNpQADdaCizpdobO6TJRAt7QNw8WbtBOozCFkTFJSFCKwjq4an1QOFaAK0g60UUlPxNY0CA)

**Comment by samouri** [comment]
```ts
// @filename: other.ts
export class A {}

// @filename: index.ts
import type { A } from './other';

// This should throw the error: "'A' cannot be used as a value because it was imported using 'import type'"
const object = { A }; 
```

**Comment by samouri** [comment]
Thank you for fixing this so quickly!
Is there any way for me to track which release this makes it into?

**Comment by RyanCavanaugh** [comment]
@samouri this will be in TypeScript 4.4 (the RC and final, but not the beta which has already passed).
